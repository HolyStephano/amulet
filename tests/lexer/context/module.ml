(* Implicit begin *)
module X =
  let _ = 1

(* Explicit begin *)
module X = struct
  let _ = 1
end

(* Explicit begin with weird alignment *)
module X =
struct
  let _ = 1
end

module X =
  struct
    let _ = 1
  end

module X = Y

(* Access modifiers *)
private module Y =
  let a = 0

(* Module types *)
module type Y = sig
 type t
 val x : t
end

module type Z = (Arg : Y) -> sig val y : Arg.t end

(* Functors *)
module X = fun (Arg : Y) struct
 let y = Arg.x
end

(* Ascriptions *)
module X' = X : Z

open X Arg
