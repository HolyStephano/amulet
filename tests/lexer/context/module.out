
module X =
  $begin let _ = 1


$end $end $sep module X = struct
  let _ = 1
$end end


$sep module X =
struct
  let _ = 1
$end end

$sep module X =
  struct
    let _ = 1
  $end end

$sep module X = Y


$sep private module Y =
  $begin let a = 0


$end $end $sep module type Y = sig
 type t
 $sep val x : t
end

$sep module type Z = ( Arg : Y ) -> sig val y : Arg.t end


$sep module X = fun ( Arg : Y ) struct
 let y = Arg.x
$end end


$sep module X' = X : Z

$sep open X Arg
<eof>
